src/JustSaying/JustSayingBus.cs

```diff
@@ -131,26 +130,26 @@ public async Task StartAsync(CancellationToken stoppingToken)
             if (stoppingToken.IsCancellationRequested) return;
 
             // Double check lock to ensure single-start
-            if (!_busStarted)
+            if (_startupTasks.Count > 0)
             {
                 await _startLock.WaitAsync(stoppingToken).ConfigureAwait(false);
                 try
                 {
-                    if (!_busStarted)
+                    if (_startupTasks.Count > 0)
```

Comment:
This no longer protects from multiple-entrants, as multiple calls to this could execute in different threads. Perhaps we need to check for both `_busStarted` and that there are startup tasks? If there are none we could mark as started.

Reply:
Good spot, I've reverted this and will go again with the suggestion you made